src/test/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantSourceTest.java

```diff
@@ -0,0 +1,165 @@
+        }
+        // Now make a PCollection, to verify that variants can be coded.
+        Pipeline p = TestPipeline.create();
+
+        // Now, we can test that we can get a PCollection from a file
+        List<String> files = Arrays.asList(FEATURE_DATA_SOURCE_TEST_DIRECTORY + "feature_data_source_test.vcf");
+        VariantsDataflowSource variantsDataflowSource = new VariantsDataflowSource(files, p);
+        PCollection<Variant> allVariants = variantsDataflowSource.getAllVariants();
+
+        // We have to clear the UUIDs to make the comparison.
+        PCollection<Variant> allVariants2 = allVariants.apply(ParDo.of(new clearUUIDDoFn()));
+        DataflowAssert.that(allVariants2).containsInAnyOrder(variants);
+        p.run();
+    }
+
+    @Test(dataProvider = "VariantDataProvider")
+    public void testVariantAdapter(final List<Variant> expectedVariantList) {
+        // The test suite for reading in VCF files is FeatureDataSourceUnitTest.
+        try (FeatureDataSource<VariantContext> featureSource = new FeatureDataSource<>(QUERY_TEST_VCF, new VCFCodec())) {
+            List<Variant> variantList = new ArrayList<>();
+            for (VariantContext feature : featureSource) {
+                VariantContextVariantAdapter va = new VariantContextVariantAdapter(feature);
+                variantList.add(va);
+            }
+            // Now, test to see that every variant is in in the expected set.
+            Assert.assertEquals(variantList.size(), expectedVariantList.size());
+            for (Variant v : variantList) {
+                boolean matchFound = false;
+                for (Variant vv : expectedVariantList) {
+                    if (VariantUtils.variantsAreEqual(v, vv)) {
+                        matchFound = true;
+                    }
+                }
+                Assert.assertTrue(matchFound, v.toString());
+            }
+        }
+    }
+
+    @Test(dataProvider = "FileVariantDataProvider")
+    public void coderTest(final List<Variant> variantList) {
+        // The simplest way to figure out if a class is coded correctly is to create a PCollection
+        // of that type and see if matches the List version.
+        Pipeline p = TestPipeline.create();
+        // Turns out we don't need to register a coder for VariantContextVariantAdapter.
+
+        PCollection<Variant> pShards = p.apply(Create.of(variantList));
+
+        List<Variant> sameVariants = Lists.newArrayList();
+        Assert.assertTrue(sameVariants.addAll(variantList));
+        DataflowAssert.that(pShards).containsInAnyOrder(sameVariants);
+        p.run();
+    }
+
+
+    @DataProvider(name = "VariantDataProvider")
+    public Object[][] getVariantData() {
+        List<Variant> variantSet = new ArrayList<>();
+        variantSet.add(new SkeletonVariant(new SimpleInterval("1", 100, 100), true, false, defaultUUID()));
```

Comment:
Are these identical to the`VariantDataProvider` in `VariantContextVariantAdapterTest`  If yes, please de-duplicate.